### PR TITLE
Show edit profile

### PR DESCRIPTION
**Pull request by MirkaaStefanov**

### CODE REVIEW

File: src/main/java/com/example/ludogoriesoft/lukeriaerpapi/services/UserService.java [hunk 1]
```diff
@@ -64,38 +83,78 @@ public User findByEmail(String email) {
...
+
+    public boolean ifPasswordMatch(String password) {
+        UserDTO authenticatedUserDTO = findAuthenticatedUser();
+        BCryptPasswordEncoder passwordEncoder = new BCryptPasswordEncoder();
+        boolean ifMatch = passwordEncoder.matches(password, authenticatedUserDTO.getPassword());
+        if (ifMatch) {
+            return true;
+        } else {
+            return false;
         }
```
Comment (berki0) [inline src/main/java/com/example/ludogoriesoft/lukeriaerpapi/services/UserService.java:129]: you can shorten the method significantly like this -**_public boolean ifPasswordMatch(String password) {
    UserDTO authenticatedUserDTO = findAuthenticatedUser();
    BCryptPasswordEncoder passwordEncoder = new BCryptPasswordEncoder();
    return passwordEncoder.matches(password, authenticatedUserDTO.getPassword());_**
}

File: src/main/java/com/example/ludogoriesoft/lukeriaerpapi/config/SecurityConfiguration.java [hunk 2]
```diff
@@ -48,7 +48,11 @@ public SecurityFilterChain securityFilterChain(HttpSecurity http) throws Excepti
                         "/api/v1/images/**",
                         "/api/v1/auth/**")
                 .permitAll()
-
+                .requestMatchers(HttpMethod.GET,"/api/v1/user/me").authenticated()
+                .requestMatchers(HttpMethod.GET,"/api/v1/user/{id}").authenticated()
+                .requestMatchers(HttpMethod.PUT,"/api/v1/user/authenticated/{id}").authenticated()
+                .requestMatchers(HttpMethod.GET,"/api/v1/user/ifPassMatch").authenticated()
+                .requestMatchers(HttpMethod.PUT,"/api/v1/user/change-pass").authenticated()
```
Comment (berki0) [inline src/main/java/com/example/ludogoriesoft/lukeriaerpapi/config/SecurityConfiguration.java:55]: You can change every thing with .requestMatchers("/api/v1/user/**").authenticated()

File: src/main/java/com/example/ludogoriesoft/lukeriaerpapi/services/UserService.java [hunk 3]
```diff
@@ -64,38 +83,78 @@ public User findByEmail(String email) {
...
+        return AuthenticationResponse
+                .builder()
+                .accessToken(jwtToken)
+                .refreshToken(refreshToken)
+                .user(modelMapper.map(updatedUser, PublicUserDTO.class))
+                .build();
+
+
+    }
+
```
Comment (berki0) [inline src/main/java/com/example/ludogoriesoft/lukeriaerpapi/services/UserService.java:110]: you can break the method into smaller parts

File: src/main/java/com/example/ludogoriesoft/lukeriaerpapi/config/SecurityConfiguration.java (hunk 2 above)
Comment (MirkaaStefanov) [reply src/main/java/com/example/ludogoriesoft/lukeriaerpapi/config/SecurityConfiguration.java:55]: but I think that only admin should be able to access this:@PutMapping("/{id}")
    public ResponseEntity<UserDTO> updateUser(@PathVariable("id") Long id, @Valid @RequestBody UserDTO userDTO, @RequestHeader("Authorization") String auth).  and this:@GetMapping
    public ResponseEntity<List<UserDTO>> getAllUsers()

Comment (MirkaaStefanov) [reply src/main/java/com/example/ludogoriesoft/lukeriaerpapi/config/SecurityConfiguration.java:55]: I think this methods should be only for admins:
                .requestMatchers(HttpMethod.GET,"/api/v1/user").hasAnyRole(ADMIN.name())
                .requestMatchers(HttpMethod.POST,"/api/v1/user").hasAnyRole(ADMIN.name())
                .requestMatchers(HttpMethod.PUT,"/api/v1/user/{id}").hasAnyRole(ADMIN.name())
                .requestMatchers(HttpMethod.DELETE,"/api/v1/user/{id}").hasAnyRole(ADMIN.name())

Comment (berki0) [reply src/main/java/com/example/ludogoriesoft/lukeriaerpapi/config/SecurityConfiguration.java:55]: ok you can do it this way -.requestMatchers("/api/v1/user/**").hasAnyRole(ADMIN.name()) (when you don't declare HTTPMETOD , default to all), also no need to declare url after "/user" use **

**Review by GeorgiStoyanovAtanasov — APPROVED** [review]

**Pull request merged**